src/test/java/com/github/digitopolis/echoserver/socket/SocketCreatorTest.java

```diff
@@ -0,0 +1,42 @@
+package com.github.digitopolis.echoserver.socket;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import com.github.digitopolis.echoserver.EchoServer;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class SocketCreatorTest {
+    @Test
+    public void testCreatesServerSocket() throws IOException {
+        ServerSocket testSocket = SocketCreator.createServerSocket(8080);
+        assertNotNull(testSocket);
+        testSocket.close();
+    }
+
+    @Test
+    public void testCreatesSocketAtGivenPort() throws IOException {
+        ServerSocket testSocket = SocketCreator.createServerSocket(5555);
+        assertEquals(testSocket.getLocalPort(), 5555);
+        testSocket.close();
```

Comment:
Similar comment here - a test for the sad path, i.e, the port is taken and the socket cannot connect, or some general socket error.